core/templates/dev/head/components/state-directives/outcome-editor/outcome-editor.directive.ts

```diff
@@ -178,6 +178,9 @@ angular.module('oppia').directive('outcomeEditor', [
             if (ctrl.savedOutcome.dest === ctrl.outcome.dest) {
               ctrl.savedOutcome.dest = StateEditorService.getActiveStateName();
             }
+            if (StateEditorService.isInQuestionMode()) {
```

Comment:
Logically the previous clause (lines 175-180) should not run in question mode, so consider putting it in an "else" instead to make clear that these are mutually exclusive decision paths.

Reply:
Actually, an issue was that the other clause ran because both were null (I think). So, if it is in the question editor, I am overriding it. So, else wouldn't work here.
Also, `getActiveStateName` returned 'question', which led it to not being null anymore which led to the backend validation error.

Reply:
Wait, I mean:

    if (StateEditorService.isInQuestionMode()) {
      ... 
   } else if (ctrl.SavedOutcome.dest === ctrl.outcome.dest) {
      ...
   }

Wouldn't that work?

Reply:
Oh, I thought you meant the other way around. Yeah, this would work!

Reply:
Done